src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -523,6 +526,13 @@ private HttpMessageHandlerStage SetupHandlerChain()
             return _handler;
         }
 
+        // Allows for DecompressionHandler (and its compression dependencies) to be trimmed when
+        // AutomaticDecompression is not being used.
+        private void EnsureDecompressionHandlerFactory()
+        {
+            _decompressionHandlerFactory ??= (settings, handler) => new DecompressionHandler(settings._automaticDecompression, handler);
```

Comment:
Nit: just to make sure we don't accidentally create a closure
```suggestion
            _decompressionHandlerFactory ??= static (settings, handler) => new DecompressionHandler(settings._automaticDecompression, handler);
```